### PR TITLE
Fixes #8720: Generate a json file containing node properties for each node

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://www.rudder-project.org/redmine/issues/8720

### CODE REVIEW

File: rudder-core/src/main/scala/com/normation/rudder/services/policies/RudderPromiseWriterServiceImpl.scala [hunk 1]
```diff
@@ -313,6 +325,31 @@ class RudderCf3PromisesFileWriterServiceImpl(
...
+      import net.liftweb.json.JsonDSL._
+      import com.normation.rudder.domain.nodes.JsonSerialisation._
+      ( "node-properties" -> properties.toDataJson())
+    }
+
+    val fileName = "node-properties.json"
+    val path = outPath + "/properties.d"
+    val jsonProperties = generateNodePropertiesJson(node.nodeInfo.properties)
+    val propertyContent = Printer.pretty(JsonAST.render(jsonProperties))
+    logger.trace(s"Create Node properties  file '${path}/${fileName}'")
```
Comment (matya) [inline rudder-core/src/main/scala/com/normation/rudder/services/policies/RudderPromiseWriterServiceImpl.scala]: There are two spaces between **properties** and **file**

Comment (fanf) [inline rudder-core/src/main/scala/com/normation/rudder/services/policies/RudderPromiseWriterServiceImpl.scala]: +1

File: rudder-core/src/main/scala/com/normation/rudder/services/policies/RudderPromiseWriterServiceImpl.scala [hunk 2]
```diff
@@ -313,6 +325,31 @@ class RudderCf3PromisesFileWriterServiceImpl(
...
+  private[this] def writeNodePropertiesFile (node: NodeConfiguration, outPath: String) = {
+
+    def generateNodePropertiesJson(properties : Seq[NodeProperty]): JValue = {
+      import net.liftweb.json.JsonDSL._
+      import com.normation.rudder.domain.nodes.JsonSerialisation._
+      ( "node-properties" -> properties.toDataJson())
+    }
+
+    val fileName = "node-properties.json"
+    val path = outPath + "/properties.d"
```
Comment (fanf) [inline rudder-core/src/main/scala/com/normation/rudder/services/policies/RudderPromiseWriterServiceImpl.scala]: This two things are sufficiently remarkable to be in parameter, and have a constant defined for them.

Comment (fanf) [inline rudder-core/src/main/scala/com/normation/rudder/services/policies/RudderPromiseWriterServiceImpl.scala]: You should not use "/" but FileSeparator. 
Or new File(new File(outPath, propertiesDir), fileName)

**Comment by fanf** [comment]
Appart for the little remarks, it seems good.

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
OK, merging this PR

**Pull request merged**